pephub/routers/eido/eido.py

```diff
@@ -158,176 +47,22 @@ async def get_schema(request: Request, namespace: str, project: str):
-            _ = pd.read_csv(StringIO(sample_table))
-    except yaml.YAMLError as e:
-        raise HTTPException(status_code=406, detail={error_key_name: str(e)})
-    except pd.errors.ParserError as e:
-        raise HTTPException(status_code=406, detail={error_key_name: str(e)})
-
-    # cleanup any existing temp dir
-    try:
-        shutil.rmtree(tmpdirname)
-    except FileExistsError:
-        pass  # ignore if dir doesn't exist
-    except FileNotFoundError:
-        pass  # ignore if dir doesn't exist
-    # save project config and sample table to temp dir
-    try:
-        os.mkdir(tmpdirname)
-        project_path = f"{tmpdirname}/project_config.yaml"
-        with open(project_path, mode="w") as f:
-            f.write(json.dumps(project_config_dict))
-
-        # write sample table to temp dir if exists
-        if sample_table is not None:
-            sample_path = f"{tmpdirname}/{project_config_dict['sample_table']}"
-            with open(sample_path, mode="w") as f:
-                f.write(sample_table)
-        try:
-            project = peppy.Project(f"{tmpdirname}/project_config.yaml")
-        except Exception as e:
-            raise HTTPException(status_code=406, detail={error_key_name: str(e)})
-    finally:
-        # delete the temp dir
-        shutil.rmtree(tmpdirname)
-    try:
-        eido.validate_config(
-            project,
-            "http://schema.databio.org/pep/2.0.0.yaml",  # just use the base PEP schema for now
-            exclude_case=True,
-        )
-    except Exception as e:
-        raise HTTPException(status_code=406, detail={error_key_name: str(e)})
-
-    # validate samples if given
-    if sample_table is not None:
-        for sample in project.samples:
-            try:
-                eido.validate_sample(
-                    project,
-                    sample.sample_name,
-                    "http://schema.databio.org/pep/2.0.0.yaml",  # just use the base PEP schema for now
-                    exclude_case=True,
-                )
-            except Exception as e:
-                return HTTPException(status_code=406, detail={error_key_name: str(e)})
-
-    return True
-
-
-@router.get("/")
-async def main():
-    print(je.list_templates())
-    return HTMLResponse(je.get_template("hello.html").render())
-
-
-# @router.get("/")
-# async def main():
-#     return HTMLResponse(je.get_template("schema.html").render())
-
-
-@router.get("/validator")
-async def main():
-    print(je.list_templates())
-    return FileResponse(os.path.join(STATICS_PATH, "index.html"))
-
-
-@router.get("/validator")
-async def main():
-    print(je.list_templates())
-    return FileResponse(os.path.join(STATICS_PATH, "schemas.html"))
-
-
 # NEW STUFF FOR REACT FRONTEND
 @router.post("/validate")
 async def validate(
     # accept both pep_registry and pep_files, both should be optional
     pep_registry: Optional[str] = Form(None),
     pep_files: Optional[List[UploadFile]] = None,
```

Comment:
Do we have an option for sending a PEP as a string?

Reply:
I don't think unless you consider pep registry as sending a pep as a string, that was what I was thinking if I had time was to create a pep_paste for sending a PEP as a string so we had the capability of pasting for validation